src/hooks/useMutationObserver/useMutationObserver.test.tsx

```diff
@@ -0,0 +1,48 @@
+import { jest } from '@jest/globals';
+import { renderHook } from '@testing-library/react';
+import { createMicrotask } from '../../index.js';
+import { useMutationObserver } from './useMutationObserver.js';
+
+describe('useMutationObserver', () => {
+  it('should observe mutations on the target element', async () => {
+    const target = document.createElement('div');
+    const callback = jest.fn();
+
+    renderHook(() => {
+      useMutationObserver(target, callback, { childList: true });
+    });
+
+    target.append(document.createElement('span'));
+
+    await createMicrotask();
```

Comment:
Can't you wrap the expect in a `await waitFor` ?

Reply:
`waitFor` polls for a certain amount of time and isn't as accurate/quick compared to microtasks